src/15_classes.py

```diff
@@ -1,29 +1,57 @@
 # Make a class LatLon that can be passed parameters `lat` and `lon` to the
 # constructor
 
-# YOUR CODE HERE
-        
-# Make a class Waypoint that can be passed parameters `name`, `lat`, and `lon` to the
-# constructor. It should inherit from LatLon. Look up the `super` method.
 
-# YOUR CODE HERE
+class LatLon:
+    def __init__(self, lat, lon):
+        self.lat = lat
+        self.lon = lon
+
+# Make a class Waypoint that can be passed parameters `name`, `lat`, and `lon` to
+# the constructor. It should inherit from LatLon. Look up the `super` method.
+
+
+class Waypoint(LatLon):
+    def __init__(self, lat, lon, name):
+        super().__init__(lat, lon)
+        self.name = name
+
+    def __str__(self):
+        return f'\tName: {self.name}, Lat: {self.lat}, Lon: {self.lon}'
 
 # Make a class Geocache that can be passed parameters `name`, `difficulty`,
 # `size`, `lat`, and `lon` to the constructor. What should it inherit from?
 
-# YOUR CODE HERE
+
+class Geocache(Waypoint):
+    def __init__(self, lat, lon, name, difficulty, size):
+        super().__init__(lat, lon, name)
+        self.difficulty = difficulty
+        self.size = size
+
+    def __str__(self):
+        return f'''
+        Name: {self.name},
+        Lat: {self.lat},
+        Lon: {self.lon},
+        Difficulty: {self.difficulty},
+        Size: {self.size}
+        '''
 
 # Make a new waypoint and print it out: "Catacombs", 41.70505, -121.51521
 
-# YOUR CODE HERE
+
+waypoint = Waypoint(name="Catacombs", lat=41.70505, lon=-121.51521)
+print('')
 
 # Without changing the following line, how can you make it print into something
 # more human-readable? Hint: Look up the `object.__str__` method
 print(waypoint)
 
 # Make a new geocache "Newberry Views", diff 1.5, size 2, 44.052137, -121.41556
 
-# YOUR CODE HERE
+geocache = Geocache(name="Newberry Views", difficulty=1.5,
+                    size=2, lat=44.052137, lon=-121.41556)
```

Comment:
I assume you already know this, but if you put the values in order as assigned in the constructor you can get rid of all the variable names in the function call... 
ex) Geocache(44.05, -121.42, "Newberry Views", 1.5, 2)
i wouldn't say either way is better than the other, they both have their trade offs for readability and all that.

Reply:
I felt like I kind of had to do it with lat/lon and by then when I was doing that, it was downhill from there... I thought to myself "I might as well specify everything, right?"